pywemo/subscribe.py

```diff
@@ -35,32 +73,124 @@ def get_ip_address(host='1.2.3.4'):
 
 
 class RequestHandler(BaseHTTPRequestHandler):
-    """Handles subscription responses received from devices."""
-
-    def do_NOTIFY(self):
+    """Handles subscription responses and long press actions from devices.
+
+    Subscription responses:
+      Pywemo can subscribe to Wemo devices. When subscribed, the Wemo device
+      will send notifications when the state of the device changes. The
+      do_NOTIFY method below is called when a Wemo device changes state.
+
+    Long press actions:
+      Wemo devices can control the state of other Wemo devices based on the
+      rules configured for the device. A long press rule is activated whenever
+      the button on the Wemo device is pressed for 2 seconds. The long press
+      rule is meant to be used to control the state of another device (turn
+      on/off/toggle). However for pywemo's use, a long press rule can be used to
+      trigger an event notification. This is implemented by configuring the
+      Wemo device to "control the state" of a virtual Wemo device. The virtual
+      device is implemented by this class.
+
+      The do_GET/do_POST/do_SUBSCRIBE methods below implement a virtual Wemo
+      device. The virtual device receives requests to change its state from
+      other Wemo devices on the network. When a Wemo device is configured to
+      change the state of the virutal device via a long press rule the following
+      sequence occurs:
+
+      1. The Wemo device will attempt to locate the virtual device on the
+      network. This is handled by the pywemo.ssdp.DiscoveryResponder class. See
+      the documentation there for more information about this step.
+
+      2. The Wemo device will fetch /setup.xml from do_GET to learn of the
+      virtual device details.
+
+      3. The Wemo device will subscribe to BinaryState notifications from the
+      virtual device. The virtual device does not send any BinaryState
+      notifications, but this step seems to be necessary before the next step
+      can happen. This step is implemented by the do_SUBSCRIBE method.
+
+      4. When a person presses the button on the Wemo for 2 seconds a long press
+      rule is triggered. If the long press rule is configured with an action
+      for the virtual device, the Wemo device will then call the do_POST method
+      to update the BinaryState of the virtual device. This doesn't actually
+      update any state, rather the virtual device then delivers the event
+      notification to any event listeners configured to receive events from the
+      pywemo SubscriptionRegistry. The event type for a long press action is
+      EVENT_TYPE_LONG_PRESS.
+    """
+
+    def do_NOTIFY(self):  # pylint: disable=invalid-name
```

Comment:
Remove the local pylint disable invalid-name here as you've added it to the pylintrc file (I agree it should be there)

Reply:
Ah, I see I looked at this change backwards.  You've removed `do_NOTIFY` and added this local disable here.  I'm fine either way then.  Perhaps the local disable makes sense since it is only defined once.  Then just use the global list for names that may be used more frequently.  But I'm okay either way --- adding them to pylintrc or just doing it locally.

Reply:
>  Perhaps the local disable makes sense since it is only defined once. Then just use the global list for names that may be used more frequently.

This sounds good to me.

When I added the three new methods, I was surprised by the fact that pylint was complaining about them. It took me some time to figure out why pylint complained about `do_GET`, `do_SUBSCRIBE` & `do_POST` but not `do_NOTIFY`. I'd err on the side of putting fewer things inside pylintrc to avoid surprising behavior.